pydicom/dataset.py

```diff
@@ -355,78 +362,164 @@ def __iter__(self):
+            if self.is_little_endian != sys_is_little_endian:
+                numpy_format.newbyteorder('S')
+
+            # Have correct Numpy format, so create the NumPy array
+            pixel_array = numpy.fromstring(self.PixelData, numpy_format)
+            
+        else:
+            # if the transfer syntax is not supported, we try to fall back to GDCM
+            if not have_gdcm:
+                msg = "The GDCM package is required to use pixel_array for transfer syntaxes that pydicom does not support natively, and GDCM could not be imported.\n"
+                raise ImportError(msg)
+            if not self.filename:
+                msg = "GDCM needs the filename to read."
+                raise TypeError(msg)
+            
+            # read the file using GDCM
+            gdcm_image_reader = gdcm.ImageReader()
+            gdcm_image_reader.SetFileName(self.filename)
+            if not gdcm_image_reader.Read():
+                raise TypeError("GDCM could not read DICOM image")
+            gdcm_image = gdcm_image_reader.GetImage()
+            
+            # determine the correct numpy datatype
+            gdcm_numpy_typemap = {
+                gdcm.PixelFormat.INT8:     numpy.int8,
+                gdcm.PixelFormat.UINT8:    numpy.uint8,
+                gdcm.PixelFormat.UINT16:   numpy.uint16,
+                gdcm.PixelFormat.INT16:    numpy.int16,
+                gdcm.PixelFormat.UINT32:   numpy.uint32,
+                gdcm.PixelFormat.INT32:    numpy.int32,
+                gdcm.PixelFormat.FLOAT32:  numpy.float32,
+                gdcm.PixelFormat.FLOAT64:  numpy.float64
+            }
+            gdcm_pixel_format = gdcm_image.GetPixelFormat().GetScalarType()
+            if gdcm_pixel_format in gdcm_numpy_typemap:
+                numpy_dtype = gdcm_numpy_typemap[gdcm_pixel_format]
+            else:
+                raise TypeError('{} is not a GDCM supported pixel format'.format(gdcm_pixel_format))
+            
+            # get the raw data buffer (decompressed, decoded)
+            gdcm_raw_data_buffer = None
+            
+            # in the case of a lookup table try to apply it
+            if gdcm_image.GetPhotometricInterpretation().GetType() == gdcm.PhotometricInterpretation.PALETTE_COLOR:
+                gdcm_buf_len = gdcm_image.GetBufferLength()
+                # apply lookup table if possible
+                gdcm_lutfilt = gdcm.ImageApplyLookupTable();
+                gdcm_lutfilt.SetInput(gdcm_image);
+                if not gdcm_lutfilt.Apply():
+                    warnings.warn("GDCM failed to apply LUT")
+                gdcm_lutfiltered_pixmap = gdcm_lutfilt.GetOutputAsPixmap()
+                gdcm_raw_data_buffer = gdcm_lutfiltered_pixmap.GetBuffer()
+                
+                # amount of samples per pixel can change
+                self.SamplesPerPixel = gdcm_lutfiltered_pixmap.GetBufferLength() / gdcm_buf_len
+                self.PlanarConfiguration = 0
```

Comment:
I don't like how I overwrite `self.SamplesPerPixel` and `self.PlanarConfiguration` in the case of a LUT. It feels like I might be corrupting internal state.

Do you guys have any suggestions to handle this more elegantly?